### PR TITLE
[KernelGen][Nvidia] Add moe_gate_top_k_routing operator with Triton kernel

### PR DESCRIPTION
**Pull request by Dingxingdi**
## Summary

Adds a Triton kernel implementation for `moe_gate_top_k_routing`.

The operator applies softmax to MoE gate logits and selects the top-k experts for each token, returning top-k weights and expert indices.

## Testing

- `CUDA_VISIBLE_DEVICES=2 python -m pytest tests/test_moe_gate_top_k_routing.py -m moe_gate_top_k_routing -vs`
- `CUDA_VISIBLE_DEVICES=2 python -m pytest benchmark/test_moe_gate_top_k_routing.py -m moe_gate_top_k_routing -s --level core`
- `pre-commit run --files ...`
- `check_operator.py --strict`
- `check_overload_consistency.py`

## Correctness

NVIDIA H20 accuracy test passed:

- 81 test cases passed
- dtypes: `float16`, `float32`, `bfloat16`

## Performance

NVIDIA H20 benchmark passed:

| Backend | Cases | Mean Speedup |
|---|---:|---:|
| NVIDIA H20 | 30 | 2.792x |

Benchmark covers `float16`, `float32`, and `bfloat16` over core MoE gate routing shapes.

## Multi-backend Testing

| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---:|---|
| NVIDIA H20 | PASS | PASS | 2.792x | Primary validation |
| Tianshu | PASS | PASS | 6.654x | Existing backend result |
| Muxi | FAIL | N/A | - | Existing backend result |
| Ascend | FAIL | N/A | - | Existing backend result |
| Hygon | FAIL | PASS | 5.274x | Existing backend result |

## Files Changed

- `src/flag_gems/ops/moe_gate_top_k_routing.py`
- `tests/test_moe_gate_top_k_routing.py`
- `benchmark/test_moe_gate_top_k_routing.py`
- `src/flag_gems/ops/__init__.py`
- `src/flag_gems/__init__.py`
- `conf/operators.yaml`

### CODE REVIEW

File: conf/operators.yaml [hunk 1]
```diff
@@ -4980,10 +4443,22 @@ ops:
...
+    stages:
+      - beta: '4.0'
+      - stable: '4.2'
+  - id: moe_gate_top_k_routing
+    description: |
+      Applies softmax to MoE gate logits and selects the top-k experts for each token.
+    for:
+      - moe_gate_top_k_routing
+    labels:
+      - aten
```
Comment (tengqm) [inline conf/operators.yaml]: This is not an aten operator. Please move the implementation into `src/flag_gems/fused` if so.

Comment (Dingxingdi) [reply conf/operators.yaml]: Fixed.

File: src/flag_gems/fused/moe_gate_top_k_routing.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    Args:
+        gate_logits: Gate logits tensor of shape [num_tokens, num_experts]
+        topk: Number of top experts to select
+
+    Returns:
+        Tuple of (topk_weights, topk_indices):
+            - topk_weights: Weights for selected experts, shape [num_tokens, topk]
+            - topk_indices: Indices of selected experts, shape [num_tokens, topk]
+    """
+    logger.debug("GEMS moe_gate_top_k_routing")
```
Comment (tengqm) [inline src/flag_gems/fused/moe_gate_top_k_routing.py]:
```suggestion
    logger.debug("GEMS MOE_GATE_TOP_K_ROUTING")
```

File: tests/test_moe_gate_top_k_routing.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+@pytest.mark.moe_gate_top_k_routing
+@pytest.mark.parametrize("num_tokens", [1, 16, 128])
+@pytest.mark.parametrize("num_experts", [4, 8, 16])
+@pytest.mark.parametrize("topk", [1, 2, 4])
+@pytest.mark.parametrize("dtype", utils.FLOAT_DTYPES)
+def test_moe_gate_top_k_routing(num_tokens, num_experts, topk, dtype):
+    if topk > num_experts:
+        pytest.skip("topk must be <= num_experts")
```
Comment (tengqm) [inline tests/test_moe_gate_top_k_routing.py:48]: For invalid input parameter combinations, simply return here would be okay.
We use `pytest.skip` for known defects in the backend or operator.
